library/core/src/panicking.rs

```diff
@@ -56,6 +56,16 @@ pub const fn panic_str(expr: &str) -> ! {
     panic_display(&expr);
 }
 
+#[cfg(not(bootstrap))]
+#[inline]
+#[track_caller]
+#[lang = "unreachable_display"] // needed for `non-fmt-panics` lint
+#[rustc_do_not_const_check] // hooked by const-eval
+#[rustc_const_unstable(feature = "core_panic", issue = "none")]
+pub const fn unreachable_display<T: fmt::Display>(x: &T) -> ! {
```

Comment:
`unreachable!(arg)` has never worked in const context, and the comment `// hooked by const-eval` is not true, making `rustc_do_not_const_check` problematic. I don't think there's any need to make this function `const`.